utils/bump_version.py

```diff
@@ -1,27 +1,41 @@
 #!/usr/bin/env python3
 
+from __future__ import annotations
+
 import argparse
-import os
 import re
 import sys
 import time
 from contextlib import contextmanager
+from pathlib import Path
+from typing import TYPE_CHECKING, Sequence, Union
+
+from typing_extensions import TypeAlias
 
-script_dir = os.path.dirname(__file__)
-package_dir = os.path.abspath(os.path.join(script_dir, '..'))
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+script_dir = Path(__file__).parent
+package_dir = script_dir.parent
 
 RELEASE_TYPE = {'a': 'alpha', 'b': 'beta'}
 
+VersionInfo: TypeAlias = Union[tuple[int, int, int, str], tuple[int, int, int, str, int]]
```

Comment:
From what I see, the `tuple[int, int, int, str]` form is never used, so we can just have `tuple[int, int, int, str, int]`.